tools/scanner/scan.mk

```diff
@@ -0,0 +1,66 @@
+# Copyright 2020-2022 The MLPerf Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+# Note:
+# SonarScanner requires a clean build for every run. Do not cache build files.
+
+ifdef SONAR_OUT_DIR
+	scanner_docker_args=\
+		--env SONAR_OUT_DIR=${SONAR_OUT_DIR} \
+		--env SONAR_TOKEN=${SONAR_TOKEN} \
+		--env PR_NUMBER=${PR_NUMBER} \
+		--env PR_BRANCH=${PR_BRANCH} \
+		--env PR_BASE=${PR_BASE}
+
+	# https://docs.sonarcloud.io/advanced-setup/languages/c-c-objective-c/#building-with-bazel
+	sonar_bazel_startup_options=--batch
+	sonar_bazel_build_args=--spawn_strategy=local --strategy=Genrule=local
+endif
+
+output/docker_mlperf_scanner.stamp: flutter/android/docker/image tools/scanner/Dockerfile
+	docker image build -t mlcommons/mlperf_mobile_flutter tools/scanner
+	touch $@
+
+.PHONY: scanner/image
+scanner/image: output/docker_mlperf_scanner.stamp
+
+.PHONY: scanner/build
+scanner/build:
+	bazel clean
+	build-wrapper-linux-x86-64 --out-dir "${SONAR_OUT_DIR}" \
+		make flutter/android
+  
+# TODO (anhappdev): Use MLCommons organization
+.PHONY: scanner/scan
+scanner/scan:
+	sonar-scanner \
```

Comment:
What code issues do we scan for?
How useful is it?
Since the CI passed without any changes to our current codebase I suspect that there are not many checks?

Reply:
> What code issues do we scan for?
How useful is it?

Please see https://github.com/mlcommons/mobile_app_open/issues/488#issuecomment-1236634456
and also their docs: https://docs.sonarcloud.io/improving/overview/

Currently, it does not support `dart`, but there is a 3rd party extension, maybe we can try it later.